pkg/controller/backup_controller.go

```diff
@@ -281,6 +290,47 @@ func (c *backupController) getLocationAndValidate(itm *api.Backup, defaultBackup
 	return backupLocation, validationErrors
 }
 
+// getVolumeLocationAndValidate ensures:
+// - each VSL name in Spec VolumeSnapshotLocation exists as a VSL
+// - exactly 1 VSL per existing or default provider
+// - a given default provider's VSL name is added to the Spec VolumeSnapshotLocation if it does not exist as a VSL
+func (c *backupController) getVolumeLocationAndValidate(itm *api.Backup, defaultVSLs map[provider]string) []string {
+	var errors []string
+	perProviderVSLName := make(map[provider]string)
+	var finalVSLNamesList []string
+	for _, specVSLName := range itm.Spec.VolumeSnapshotLocations {
+		// validate each specVSLName exists as a VolumeSnapshotLocation
+		vsl, err := c.snapshotLocationLister.VolumeSnapshotLocations(itm.Namespace).Get(specVSLName)
+		if err != nil {
+			errors = append(errors, fmt.Sprintf("Error getting volume snapshot location for %s: %v", specVSLName, err))
+			continue
+		}
+
+		// ensure we end up with exactly 1 specVSLName *per provider*
+		providerVSLName := perProviderVSLName[provider(vsl.Spec.Provider)]
+		if providerVSLName != "" {
+			// if > 1 VSL name per provider as in ["aws-us-east-1" | "aws-us-west-1"] (same provider, different names)
+			if providerVSLName != specVSLName {
+				errors = append(errors, fmt.Sprintf("more than one VolumeSnapshotLocation found for provider %s: %s", vsl.Spec.Provider, specVSLName))
+				continue
+			}
+
+			perProviderVSLName[provider(vsl.Spec.Provider)] = specVSLName
```

Comment:
I think you want this line after L320 - I don't think it will ever be hit as-is because this is only executing if there's already an entry in the map for this provider, but this is the only place we'd insert into the map